src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProject.cs

```diff
@@ -1440,6 +1452,21 @@ public void ProcessFileChange(string projectSystemFilePath, string workspaceFile
                 }
             }
 
+            public void ReorderFiles(ImmutableArray<string> filePaths)
+            {
+                lock (_project._gate)
+                {
+                    if (_project._activeBatchScopes > 0)
+                    {
+                        _multipleOrderedFilePathsInBatch.Add(filePaths);
```

Comment:
if we only care about the latest, would it be clearer to not saving old ones?

Reply:
This has be re-done.